app/code/Magento/CatalogSearch/Model/Adapter/Mysql/Aggregation/DataProvider.php

```diff
@@ -29,42 +26,46 @@ class DataProvider implements DataProviderInterface
     private $eavConfig;
 
     /**
-     * @var Resource
+     * @var ScopeResolverInterface
      */
-    private $resource;
+    private $scopeResolver;
 
     /**
-     * @var ScopeResolverInterface
+     * @var AdapterInterface
      */
-    private $scopeResolver;
+    private $connection;
 
     /**
-     * @var Session
+     * @var SelectBuilderForAttributeTypePrice
      */
-    private $customerSession;
+    private $selectBuilderForAttributeTypePrice;
 
     /**
-     * @var AdapterInterface
+     * @var SelectBuilderForAttribute
      */
-    private $connection;
+    private $selectBuilderForAttribute;
 
     /**
      * @param Config $eavConfig
      * @param ResourceConnection $resource
      * @param ScopeResolverInterface $scopeResolver
-     * @param Session $customerSession
+     * @param SelectBuilderForAttributeTypePrice|null $selectBuilderForAttributeTypePrice
+     * @param SelectBuilderForAttribute|null $selectBuilderForAttribute
      */
     public function __construct(
         Config $eavConfig,
         ResourceConnection $resource,
         ScopeResolverInterface $scopeResolver,
-        Session $customerSession
```

Comment:
Fixed.

Reply:
@RomaKis 
no, you did not fix that.

Because you still have `Session $customerSession,` param which you don't use

There should not be `Session` neither in comment block nor in constructor parameter list